### ISSUE TITLE
Problem with kubelet ARM image

### ISSUE THREAD

**Issue by surajssd** [poseidon/typhoon#769]
**Description**

Problem running kubelet arm image https://github.com/poseidon/kubelet.

**Steps to Reproduce**

```console
# docker run quay.io/poseidon/kubelet:v1.18.4-arm64
standard_init_linux.go:211: exec user process caused "no such file or directory"
```

**Expected behavior**

I expected the kubelet to start, this seems that the image was not properly built for arm.

**Environment**

* Platform: Packet

* OS: 

```console
# cat /etc/os-release 
NAME="Ubuntu"
VERSION="18.04.3 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04.3 LTS"
VERSION_ID="18.04"
...
UBUNTU_CODENAME=bionic
```

* Kernel:

```console
# uname -a
Linux ams1-c1.large.arm-01 4.15.0-72-generic #81-Ubuntu SMP Tue Nov 26 12:21:09 UTC 2019 aarch64 aarch64 aarch64 GNU/Linux
```

**Possible Solution**

Build the image for ARM.

**Comment by dghubble** [comment]
Yeah, I still have some changes to make to the Kubelet build process for (potential) future architectures. Thanks for trying it, I should have it fixed soon. 

Just to clarify before someone else asks: Typhoon supports only amd64 platforms as of Jun 2020. [v1.18.4](https://github.com/poseidon/typhoon/blob/master/CHANGES.md#v1184) makes infrastructure changes to begin providing and using multi-arch manifestlist images, but only the amd64 variant is exercised or tested. Its being thought about, but no announcements of Typhoon on arm64 to date.

**Comment by dghubble** [comment]
The newer images should be executable at least. Try `quay.io/poseidon/kubelet:v1.18.4-1-g925279d` or if your host can't pull multi-arch, `quay.io/poseidon/kubelet: v1.18.4-1-g925279d-arm64`. Checked on Fedora 32 on AWS A1 (I tried Flatcar Linux alpha for ARM and AWS, but the OS failed to boot :S).

As mentioned, no promises about these images since they're not exercised by Typhoon. But should be solved.

**Comment by surajssd** [comment]
`quay.io/poseidon/kubelet:v1.18.4-1-g925279d-arm64` worked fine:

```console
# docker run quay.io/poseidon/kubelet:v1.18.4-1-g925279d-arm64
I0626 10:29:10.100020       1 server.go:417] Version: v1.18.4
I0626 10:29:10.102406       1 plugins.go:100] No cloud provider specified.
W0626 10:29:10.102547       1 server.go:560] standalone mode, no API client
E0626 10:29:10.261843       1 machine.go:331] failed to get cache information for node 0: open /sys/devices/system/cpu/cpu0/cache/index0/size: no such file or directory
E0626 10:29:10.262225       1 machine.go:331] failed to get cache information for node 1: open /sys/devices/system/cpu/cpu48/cache/index0/size: no such file or directory
W0626 10:29:10.264729       1 server.go:474] No api server defined - no events will be sent to API server.
I0626 10:29:10.264785       1 server.go:647] --cgroups-per-qos enabled, but --cgroup-root was not specified.  defaulting to /
F0626 10:29:10.265773       1 server.go:274] failed to run Kubelet: running with swap on is not supported, please disable swap! or set --fail-swap-on flag to false. /proc/swaps contained: [Filename                         Type            Size    Used    Priority /dev/sda2                               partition   2441980  0       -2]
```

And with `quay.io/poseidon/kubelet:v1.18.4-1-g925279d` got following error. I think the host cannot pull the right arch image.

```console
# docker run quay.io/poseidon/kubelet:v1.18.4-1-g925279d
Unable to find image 'quay.io/poseidon/kubelet:v1.18.4-1-g925279d' locally
v1.18.4-1-g925279d: Pulling from poseidon/kubelet
docker: no matching manifest for linux/arm64/v8 in the manifest list entries.
See 'docker run --help'.
```

**Comment by dghubble** [comment]
Upstream [client](https://github.com/containers/image/issues/898) libs have discussed relaxing the variant handling when there is a clear best match like this. That hasn't made its way into podman yet. Docker/moby may adopt similar behavior too, but its not needed. If Typhoon hypotehtically offered arm64, only podman (Fedora CoreOS) and rkt (Flatcar Linux) run the Kubelet anyway.

For now, v1.18.5 will set a variant which should work for most.

**Comment by dghubble** [comment]
The variant issue should be addressed in the v1.18.5 image.

I still have more minor changes to the plumbing lined up for future images. But I don't want to track that work here, since arm64 isn't in use in Typhoon as of today.
